### DISCUSSION TITLE
Renovate doesn't support my Dockerfile FROM Image Tag

### DISCUSSION THREAD

**Discussion by MartinX3** [renovatebot/renovate#20754] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

That the docker image tags are detected.

The tags are:
- base-devel-20230226.0.129555
- base-devel-20230219.0.127702
- base-devel-20230212.0.126025
- base-devel-20201115.0.9067
- base-devel-20201108.0.8567
- base-devel-20201102.0.8296

### What you tried so far.

My config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "dockerfile": {
    "fileMatch": [
      "(^|/|\\.)Containerfile$",
      "(^|/)Containerfile[^/]*$"
    ]
  },
  "packageRules": [
    {
      "automerge": true,
      "matchDatasources": ["docker"],
      "matchPackageNames": ["archlinux"],
      "versioning": "regex:^base-devel-(\\d{8}\\.\\d{1}\\.\\d+)$"
    }
  ]
}

```

The start of my Containerfile:
```Containerfile
FROM docker.io/archlinux:base-devel-20230226.0.129555
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Dependency docker.io/archlinux has unsupported value base-devel-20230226.0.129555
```

</details>

**Comment by viceice** [comment, marked as answer]
you need to define capture group names for regex versioning. checkout the docs.

**Reply by MartinX3** [reply]
Thank you.
Now I tried many iterations, but somehow every time the logs are telling me the version is unsupported.
That's my last try `"regex:^(?<compatibility>([\\w-]+)-)?(?<patch>\\d{8}\\.\\d{1}\\.\\d+)?$"`

**Reply by rarkins** [reply]
You need to have major, minor, patch

**Reply by MartinX3** [reply]
From the docs `major, minor, and patch: at least one of these must be provided.`
So I added `patch`
I don't understand what's missing here according to https://docs.renovatebot.com/modules/versioning/#regular-expression-versioning

**Reply by viceice** [reply]
each of those groups must be a number, so you can't match the full version to a single number. there are a lot samples in the docs.

**Reply by MartinX3** [reply]
The docs say i need only one of them. The others will be 0.
So i added `patch` because the version is just a rolling release and only patch for the whole tag is important.
I used the doc examples and I added my regex tries.

Still the logs say the docker tag version is not supported.

So the capture group can't match the entire right part of the tag?

**Reply by viceice** [reply]
yes, you need to match every part of the version as a separate named group. you can of cause only the patch part, but it needs to be a single number.

**Reply by MartinX3** [reply]
Hmmmm, thank you
But I still feel stupid
I tried `^((?<compatibility>[\\w-]+)-)?((?<major>\\d{8})\\.(?<minor>\\d{1})\\.(?<patch>\\d+))?$`
According to https://regexr.com/ I match every singe number but still the logs just say the docker tag version is unsupported

**Reply by viceice** [reply]
your rules doesn't match. you need to write the full package name for `matchPackageNames`

**Reply by MartinX3** [reply]
Full package name?
I tried
`_/archlinux`
and
`archlinux`
and
`library/archlinux`
for
https://hub.docker.com/_/archlinux

My file looks like
```
      "matchDatasources": ["docker"],
      "matchPackageNames": ["_/archlinux"],
```
and I also tried
```
      "matchDatasources": ["docker"],
      "matchPackageNames": ["library/archlinux"],
```
and
```
      "matchDatasources": ["docker"],
      "matchPackageNames": ["archlinux"],
```

In the docs I see
```
      "matchDatasources": ["docker"],
      "matchPackageNames": ["badly-versioned-docker-image"],
```

is there a fuller package name than `archlinux`, `_archlinux` or `library/archlinux`?

**Reply by MartinX3** [reply]
Now I tried `docker.io/archlinux` as the package name and it works.
But why doesn't the doc tells me to use `docker.io/<package>` as the matchPackageNames?
I thought the matchDatasources `docker` redirects to `docker.io`?

**Reply by rarkins** [reply]
@MartinX3 please read all of the following document but especially https://github.com/renovatebot/renovate/blob/main/docs/development/help-us-help-you.md#avoid-complaining-about-docs-after-getting-help

**Reply by rarkins** [reply]
The package name is whatever you have in your Dockerfile/Containerfile, and you showed this:

<img width="457" alt="image" src="https://user-images.githubusercontent.com/6311784/222954394-ca7c4529-b478-40e0-86f9-89fcbe78a594.png">
